validphys2/src/validphys/config.py

```diff
@@ -1359,45 +1358,69 @@ def load_default_default_filter_settings(self, spec):
-        Currently these limits are ``q2min`` and ``w2min``.
+        Currently these limits are ``q2min``, ``w2min``, and ``maxTau``.
+
+        Parameters
+        ----------
+        filter_defaults: dict, None
+            A mapping containing the default kinematic limits to be used when
+            filtering data (when using internal cuts).
+            Currently these limits are ``q2min``, ``w2min``, and ``maxTau``.
+        
+        Returns
+        -------
+        FilterDefaults
+            A hashable object containing the default kinematic limits to be used when
+            filtering data (when using internal cuts).
+            Currently these limits are ``q2min``, ``w2min``, and ``maxTau``.
         """
         log.warning("Overwriting filter defaults")
-        return filter_defaults
+
+        parsed_filter_defaults = FilterDefaults(
+                                            q2min = filter_defaults["q2min"] if "q2min" in filter_defaults else None, 
+                                            w2min = filter_defaults["w2min"] if "w2min" in filter_defaults else None,
+                                            maxTau = filter_defaults["maxTau"] if "maxTau" in filter_defaults else None,
+                                            )
+        return parsed_filter_defaults
 
     def produce_defaults(
         self,
         q2min=None,
         w2min=None,
         maxTau=None,
         default_filter_settings=None,
-        filter_defaults={},
+        filter_defaults=FilterDefaults,
         default_filter_settings_recorded_spec_=None,
     ):
         """Produce default values for filters taking into account the
         values of ``q2min``, ``w2min`` and ``maxTau`` defined at namespace
         level and those inside a ``filter_defaults`` mapping.
+
+        Within this function the hashable type FilterDefaults is turned into
+        a dictionary so as to allow for overwriting of the values of q2min, w2min and maxTau.
+        The dictionary is then turned back into a FilterDefaults object.
         """
-        from validphys.filters import default_filter_settings_input
+        
+        if isinstance(filter_defaults, FilterDefaults):
+            filter_defaults = filter_defaults.to_dict()
 
-        if q2min is not None and "q2min" in filter_defaults and q2min != filter_defaults["q2min"]:
-            raise ConfigError("q2min defined multiple times with different values")
-        if w2min is not None and "w2min" in filter_defaults and w2min != filter_defaults["w2min"]:
-            raise ConfigError("w2min defined multiple times with different values")
+            if q2min is not None and filter_defaults["q2min"] != q2min:
+                raise ConfigError("q2min defined multiple times with different values")
+            
+            if w2min is not None and filter_defaults["w2min"] != w2min:
+                raise ConfigError("w2min defined multiple times with different values")
```

Comment:
I think this is changing too much the behaviour of the function. I would leave this as close as possible to the previous behaviour.

Reply:
No, before it would check whether I'm passing a `filter_defaults` that is incompatible with the values of `w2min`, `q2min` etc that I'm also passing.

Now this check is only done if the `filter_defaults` I'm passing is an instance of `FilterDefaults`. Note that regardless of the default input of the function I can call it with e.g.

`produce_defaults(qmin=1, filter_defaults={"q2min": 4})`

Reply:
I think this is the only part left to be addressed to avoid having `filter_defaults` which are incompatible with the input values of `maxTau, q2min, w2min`